src/templates/charm_policy.hcl

```diff
@@ -0,0 +1,66 @@
+# Allow management of policies starting with charm- prefix
+path "sys/policy/charm-*" {
+  capabilities = ["create", "read", "update", "delete"]
+}
+# Allow operations under the charm prefix
+path "charm-*" {
+  capabilities = [ "create", "read", "update", "delete", "list", "sudo" ]
+}
+
+# Allow discovery of all policies
+path "sys/policy/" {
+  capabilities = ["list"]
+}
+
+# Allow management of approle's with charm- prefix
+path "auth/approle/role/charm-*" {
+  capabilities = ["create", "read", "update", "delete", "list"]
+}
+
+# Allow discovery of approles
+path "auth/approle/role" {
+  capabilities = ["read"]
+}
+path "auth/approle/role/" {
+  capabilities = ["list"]
+}
+
+# Allow charm- prefixes secrets backends to be mounted and managed
+path "sys/mounts/charm-*" {
+  capabilities = ["create", "read", "update", "delete", "sudo"]
+}
+
+# Allow charm- prefixes pki backends to be used
+path "charm-pki-*" {
```

Comment:
Isn't this included in the one specific for `charm-*` a few line above?

Reply:
You're right. I've removed this one.